lib/api/completeMultipartUpload.js

```diff
@@ -182,7 +182,13 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                 // in numerical order
                 // return an error
                 if (Number.parseInt(part.PartNumber[0], 10) !== i + 1) {
-                    return next(errors.InvalidPartOrder);
+                    // If not in order return InvalidPartOrder
+                    if (jsonList.Part.some(partVal =>
```

Comment:
Actually, you are totally right. Don't know how I came this messy thing. Thanks.